Core/include/Acts/Material/IntersectionMaterialAssigner.hpp

```diff
@@ -0,0 +1,70 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Detector/DetectorVolume.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Geometry/TrackingVolume.hpp"
+#include "Acts/MagneticField/MagneticFieldContext.hpp"
+#include "Acts/Material/interface/IAssignmentFinder.hpp"
+
+#include <utility>
+#include <vector>
+
+namespace Acts {
+
+/// @class IntersectionMaterialAssigner
+///
+/// A purely intersection based material assigner on a trial and error basis.
+/// This is to be used if the navigation of the propagator is not available
+/// or not reliable, or simply for cross-checking the results.
+///
+/// @note that differently to the PropagatorMaterialAssigner, this assigner
+/// needs to be preconditioned with the surfaces and volumes that are tested
+/// for candidate inclusion.
+///
+/// In a large-n material interaction scenario, this is not the most efficient
+class IntersectionMaterialAssigner final : public IAssignmentFinder {
+ public:
+  /// @brief Nested configuration struct
+  struct Config {
+    /// @brief  The surfaces to be tested
+    std::vector<const Surface*> surfaces;
+    /// @brief  The volumes to be tested: TrackingVolume
+    std::vector<const TrackingVolume*> trackingVolumes;
+    /// @brief  The volumes to be tested: DetectorVolume
+    std::vector<const Experimental::DetectorVolume*> detectorVolumes;
+  };
+
+  /// @brief Construct with the configuration
+  IntersectionMaterialAssigner(const Config& cfg) : m_cfg(cfg) {}
```

Comment:
Should we ensure somewhere (maybe in the constructor ?) That the surfaces do not belong to the tracking and detector volumes ?
I am sure we have way to catch this later on but stopping people trying to do this as soon as possible might be good ?

Reply:
Actually, we don't - you can even run this in a way that you take surfaces from inside volumes, and restrict with a veto in just an area around, and assign the rest to the volume.

Reply:
Ah yes I remember you mentioning that ! That is great.